KingPhisher

```diff
@@ -33,75 +33,75 @@
 import argparse
 import logging
 import os
+import subprocess
 import sys
-import threading
-import time
 
-if getattr(sys, 'frozen', False):
-	# set the basemap data directory for frozen builds
-	os.environ['BASEMAPDATA'] = os.path.join(os.path.dirname(sys.executable), 'mpl-basemap-data')
-
-from king_phisher import color
-from king_phisher import find
-from king_phisher import utilities
-from king_phisher import version
-from king_phisher.client import application
-from king_phisher.client import gui_utilities
-
-from gi.repository import GObject
-from gi.repository import Gtk
+from king_phisher import startup
 
 def main():
 	parser = argparse.ArgumentParser(description='King Phisher Client GUI', conflict_handler='resolve')
-	utilities.argp_add_args(parser, default_root='KingPhisher')
-	parser.add_argument('-c', '--config', dest='config_file', required=False, help='specify a configuration file to use')
-	parser.add_argument('--no-plugins', dest='use_plugins', default=True, action='store_false', help='disable all plugins')
-	parser.add_argument('--no-style', dest='use_style', default=True, action='store_false', help='disable interface styling')
-	arguments = parser.parse_args()
+	startup.argp_add_wrapper(parser)
+	startup.argp_add_client(parser)
+	startup.argb_add_default_args(parser)
+
+	arguments, unknown_args = parser.parse_known_args()
+	sys_argv = sys.argv
+	sys_argv.pop(0)
 
-	# basic runtime checks
 	if sys.version_info < (3, 4):
-		color.print_error('the Python version is too old (minimum required is 3.4)')
+		print('[-] the Python version is too old (minimum required is 3.4)')
 		return 0
-
-	if Gtk.check_version(3, 14, 0):
-		color.print_error('the GTK+ version is too old (minimum required is 3.14)')
+	logger = logging.getLogger('KingPhisher.wrapper')
+	logger.setLevel(arguments.loglvl if arguments.loglvl else 'WARNING')
+	console_log_handler = logging.StreamHandler()
+	console_log_handler.setLevel(arguments.loglvl if arguments.loglvl else 'WARNING')
+	console_log_handler.setFormatter(logging.Formatter('%(levelname)-8s %(message)s'))
+	logger.addHandler(console_log_handler)
+
+	target_directory = os.path.abspath(os.path.dirname(__file__))
+	logger.debug("target diretory: {}".format(target_directory))
+	os.environ['PIPENV_VENV_IN_PROJECT'] = os.environ.get('PIPENV_VENV_IN_PROJECT', 'True')
+	os.environ['PIPENV_PIPFILE'] = os.environ.get('PIPENV_PIPFILE', os.path.join(target_directory, 'Pipfile'))
+	logger.info('checking for pipenv for environment')
+	if not startup.which('pipenv'):
+		logger.info('installing pipenv')
+		process_output, return_code = startup.run_process([sys.executable, '-m', 'pip', 'install', 'pipenv'], cwd=target_directory)
+		if return_code:
+			logger.warning("the following issue occurred during installation of pipenv: {}".format(process_output))
```

Comment:
Since we're not continuing here, this should be an error and you should be returning non-zero. I think returning `return_code` would actually be a good idea so the error status "bubbles up".